index.js

```diff
@@ -94,6 +96,11 @@ function watchify (b, opts) {
 
     function watchFile (file, dep) {
         dep = dep || file;
+        // don't watch files which are explicitly ignored
+        if (wopts.ignored && (ignoredFiles[file] || anymatch([wopts.ignored], dep))) {
```

Comment:
A few things:

1. I think you mean `file` instead of `dep`,
2. You're remembering ignored files, cool, but you're still checking non-ignored files every time,
3. `[wopts.ignored]` is wrapping ignored arrays in arrays - which doesn't work - this would've been caught by my tests in https://github.com/substack/watchify/pull/212, except that:
4. Because you're leaving `wopts.ignored`, chokidar is still doing it's own ignore checking.

Sorry, I haven't had time to work on this. I've got a lot going on right now.